GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1687,18 +1691,11 @@ private void RevisionsCellPainting(object sender, DataGridViewCellPaintingEventA
                         int gravatarTop = e.CellBounds.Top + textHeight + 6;
                         int gravatarLeft = e.CellBounds.Left + baseOffset + 2;
 
-
-                        Image gravatar = Gravatar.GravatarService.GetImageFromCache(revision.AuthorEmail + gravatarSize.ToString() + ".png", revision.AuthorEmail, AppSettings.AuthorImageCacheDays, gravatarSize, AppSettings.GravatarCachePath, FallBackService.MonsterId);
-
-                        if (gravatar == null && !string.IsNullOrEmpty(revision.AuthorEmail))
-                        {
-                            ThreadPool.QueueUserWorkItem(o =>
-                            Gravatar.GravatarService.LoadCachedImage(revision.AuthorEmail + gravatarSize.ToString() + ".png", revision.AuthorEmail, null, AppSettings.AuthorImageCacheDays, gravatarSize, AppSettings.GravatarCachePath, RefreshGravatar, FallBackService.MonsterId));
-                        }
-
+                        var gravatar = await _gravatarService.GetAvatarAsync(revision.AuthorEmail, gravatarSize, DefaultImageType.MonsterId);
                         if (gravatar != null)
+                        {
                             e.Graphics.DrawImage(gravatar, gravatarLeft + 1, gravatarTop + 1, gravatarSize, gravatarSize);
```

Comment:
When I switch revision grid layout (ctrl-shift-L), I get an exception here:
```
System.ArgumentException was unhandled by user code
  HResult=-2147024809
  Message=Parametr jest nieprawidłowy. //Invalid argument
  Source=System.Drawing
  StackTrace:
       w System.Drawing.Graphics.CheckErrorStatus(Int32 status)
       w System.Drawing.Graphics.DrawImage(Image image, Int32 x, Int32 y, Int32 width, Int32 height)
       w GitUI.RevisionGrid.<RevisionsCellPainting>d__227.MoveNext() w C:\d\projekty\gitextensions\GitUI\UserControls\RevisionGrid.cs:wiersz 1697
  InnerException: 
```
I am not sure if awaiting inside a painting event is allowed. 
Perhaps you should call a continuation on GetAvatarAsync and invoke RefreshGravatar. It would be semantically equal to the previous implementation.